packages/frontend/src/pages/EpochKeyPage.jsx

```diff
@@ -0,0 +1,111 @@
+import React, { useContext, useEffect } from 'react'
+import { useParams, Link } from 'react-router-dom'
+import { observer } from 'mobx-react-lite'
+import state from '../contexts/state'
+import Tooltip from '../components/Tooltip'
+import EpochKeyEvent from '../components/EpochKeyEvent'
+import Footer from '../components/Footer'
+import dayjs from 'dayjs'
+import relativeTime from 'dayjs/plugin/relativeTime'
+
+dayjs.extend(relativeTime)
+
+export default observer(() => {
+  const { id } = useParams()
+  const epochKeyId = BigInt(id).toString(10)
+  // const { epochKey } = useContext(state)
+  const { unirep } = useContext(state)
+  useEffect(() => {
+    const loadData = async () => {
+      // await epochKey.loadAttestationsByEpochKey(epochKeyId)
+      !unirep.attestationsByEpochKey.has(epochKeyId)
+        ? await unirep.loadAllAttestations()
+        : null
+    }
+    loadData()
+  }, [])
+  // const attestationIds = epochKey.attestationIds
+  const attestations = unirep.attestationsByEpochKey.get(epochKeyId)
+
```

Comment:
@vimwitch I'm doing a last round of cleanup/comments before merging this + #22.
I have 2 ways to load/render data here for`EpochKeyPage.jsx `(and same for `UserPage.jsx`). Currently using only `unirep` store in both pages but could easily switch to using `epochKey` and `user` stores instead. Do you think it's better this way (remove `epochKey`, `user`: the app has fewer routes and stores overall) or switch it (load only relevant epochKey/user data instead of `unirep.loadAllxxx`, also simplify some data ingestion in `unirep.js`)?
Are there any other factors you'd consider in this choice?

Reply:
I think there's only one factor to consider, which is that the `loadAll` strategy will become impossible over time. Definitely go with loading smaller subsets of data. From a UX perspective it's better to have 10 small requests than 1 large request.

As far as logically organizing stores/routes. I tend to take different approaches with this. I have many routes, one for each bit of data i need to load. But i might have only a couple stores. I prefer to have a few stores that are a few hundred lines, rather than many stores that are a few dozen.

Something i haven't really explained about the stores is server side rendering. We can run this frontend javascript on a server when the user makes a request. So if i go to `explorer.com/attester/0x0...` i'm expecting a webpage. The server takes the react component and renders it (causing `useEffect` to be called). Once the render is complete the resulting html is sent to the user so there is no loading screen. The data from the stores is serialized to JSON and put into this html as a block of js that looks like `window.__SSR_DATA__ = JSON.stringify(state)`. Then when the html gets to the user we can `JSON.parse(window.__SSR_DATA__)` and ingest the data into the stores in the client. This is what the `load()` function is for in stores.

The key to this is, we want the smallest amount of data to be loaded during SSR, because all of that data will be sent to the client, making the response bigger and thus increasing load time. So we want a surgical approach to the data that is loaded. If i'm loading the attester page only the data that the user _immediately_ sees should be loaded.

Reply:
This pr looks good, feel free to merge when you're ready

Reply:
great, thanks for this explanation! I added the specific `user` and `epochKey` routes to `unirep.js` and removed `user` and `epochKey` contexts.